docs/modules/dev/partials/domain/external-intro.adoc

```diff
@@ -0,0 +1,12 @@
+There are a number of xref:dev:domain/external/index.adoc[external components] that, 
+although not part of the core, are essential to make the 
+James system work as a whole:
+
+ * Mailet implementations
+ * Protocol-based applications
+ * Administration tools
+ * Storage implementations
+ * Mime4j
```

Comment:
In my opinion, Mime4J and the like should be referenced as libraries and not as external components.

Reply:
To me, the vocabulary that I am used to (meaning that I am not claiming to be "right"), a "library" is something that gets embedded into a "component". Both are just ways of packaging up code, but the library is "disposable" (meaning that it is just part of the implementation and is not seem from the outside, nor cared about), while the "component" is something that is actually seen and worth understanding.

If users or maybe even developers of James don't have to care about Mime4j, then I would agree that it is a library. It would just be part of some other components, and would only come into scope when developing that component. So for this documentation project, I probably wouldn't even have to mention it at all.